auth/token-managers/jwt-token-manager.ts

```diff
@@ -188,6 +188,47 @@ export class JwtTokenManager {
     return Promise.reject(err);
   }
 
+  /**
+   * Save the JWT service response and the calculated expiration time to the object's state.
+   *
+   * @param tokenResponse - Response object from JWT service request
+   * @private
+   * @returns {void}
+   */
+  protected saveTokenInfo(tokenResponse): void {
+    const responseBody = tokenResponse.result || {};
+    const accessToken = responseBody[this.tokenName];
+
+    if (!accessToken) {
+      const err = 'Access token not present in response';
+      logger.error(err);
+      throw new Error(err);
+    }
+
+    // the time of expiration is found by decoding the JWT access token
+    // exp is the time of expire and iat is the time of token retrieval
+    const decodedResponse = jwt.decode(accessToken);
+    if (!decodedResponse) {
+      const err = 'Access token recieved is not a valid JWT'
+      logger.error(err);
+      throw new Error(err);
+    }
+
+    const { exp, iat } = decodedResponse;
+    // There are no required claims in JWT
+    if (!exp || !iat) {
+      this.expireTime = 0;
+      this.refreshTime = 0;
+    } else {
+      const fractionOfTtl = 0.8;
+      const timeToLive = exp - iat;
+      this.expireTime = exp;
+      this.refreshTime = exp - (timeToLive * (1.0 - fractionOfTtl));
+    }
+
+    this.tokenInfo = extend({}, responseBody);
+  }
```

Comment:
The methods are exactly the same, with the exception of this line: `const responseBody = tokenResponse.result || {};`, which was added. 

TS style requires the methods be listed in order of public -> protected -> private so I had to move the method, resulting in a larger-looking diff.